src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -356,6 +386,17 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary=False):
                                 "archivestatus": archStatus,
                                 "fwjr": jsonFWJR,
                                 "type": "fwjr"}
+                if countDocs >= self.fwjrdatabase.getQueueSize():
+                    try:
+                        self.fwjrdatabase.commit(callback=discardConflictingDocument)
+                    except CouchRequestTooLargeError as exc:
+                        msg = "Failed to commit bulk of framework job report to CouchDB."
+                        msg += f" Details: {str(exc)}"
+                        logging.warning(msg)
+                        shrinkLargeFJR(self.fwjrdatabase, 8e6)
+                        # now all the documents should fit in
+                        self.fwjrdatabase.commit(callback=discardConflictingDocument)
```

Comment:
here you probably need another try/except to make clear error message.

Reply:
Yes, it could be done. But we don't expect to hit that error again. So I would simply not complicate the code and let any exception to just propagate upstream.